### ISSUE TITLE
documentbuilderfactory-disallow-doctype-decl-missing doesn't work with constructors/methods

### ISSUE THREAD

**Issue by coheigea** [returntocorp/sgrep-rules#2538]
The rule documentbuilderfactory-disallow-doctype-decl-missing is being triggered incorrectly in two scenarios:

a) When a DocumentBuilderFactory instance is a class variable, and the feature is set in the constructor:

```
class GoodDocumentBuilderFactoryCtr {

    private final DocumentBuilderFactory dbf;

    public GoodDocumentBuilderFactoryCtr() throws Exception {
        dbf = DocumentBuilderFactory.newInstance();
        dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
        //ok:documentbuilderfactory-disallow-doctype-decl-missing
        dbf.newDocumentBuilder();
    }
}
```

b) When the feature is set in another method:

```
public void somemethod() throws Exception {
    DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
    setFeatures(dbf);
    //ok:documentbuilderfactory-disallow-doctype-decl-missing
    dbf.newDocumentBuilder();
}

private void setFeatures(DocumentBuilderFactory dbf) throws Exception {
    dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
}
```
In both cases I would expect it to pass. FAO @0xDC0DE

**Comment by 0xDC0DE** [comment]
Hey @coheigea ! Thanks for your report, I'll take a look!

**Comment by coheigea** [comment]
Thanks a lot @0xDC0DE , it seems to be fixed with the latest release.
